app/services/page_content.rb

```diff
@@ -0,0 +1,68 @@
+class PageContent
+  attr_accessor :book_id, :page_id, :content, :doc
+
+  def initialize(book_uuid:, book_version:, page_uuid:)
+    @book_uuid = book_uuid
+    @book_id = "#{book_uuid}@#{book_version}"
+    @page_id = "#{book_id}:#{page_uuid}"
+  end
+
+  def s3
+    @s3 ||= OpenStax::Content::S3.new
+  end
+
+  def latest_archive_version
+    @latest_archive_version ||= s3.ls.last
+  end
+
+  def overriden_archive_version
+    @overriden_archive_version ||= fetch_rex_books.dig(@book_uuid, 'archiveOverride').to_s.gsub('/apps/archive/', '').presence
+  end
+
+  def archive_version
+    overriden_archive_version || Rails.application.secrets.content[:archive_version] || latest_archive_version
```

Comment:
yea idk if i really want to be on the hook for updating this in the config every time we upgrade to a new pipeline, i know its less than ideal to do the extra queries, but the right thing is to load it from the config.json

alternatively you could switch it over to using the archive_prefix from the `source_metadata` and that part will just do nothing for now until we complete the frontend change, that might be overall less work than messing around with all these configs